### ISSUE TITLE
w2ui fails if used together with Opal

### ISSUE THREAD

**Issue by bwl21** [vitmalina/w2ui#1173]
This html-code fails. as soon as I remove opal, it works.
```html
<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="http://w2ui.com/src/w2ui-1.4.min.css" />
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
    <script type="text/javascript" src="http://w2ui.com/src/w2ui-1.4.min.js"></script>

    <script src="http://cdn.opalrb.org/opal/current/opal.js"></script>
</head>
<body>
<div id="layout" style="height: 400px"></div>
</body>
<script>
    function init_w2ui() {
        var pstyle = 'background-color: #F5F6F7; border: 1px solid #dfdfdf; padding: 5px;';
        $('#layout').w2layout({
            name: 'layout',
            panels: [
                {type: 'top', id: 'foobar', size: 50, resizable: true, style: pstyle, content: 'topsss'},
                {type: 'left', size: 200, resizable: true, style: pstyle, content: 'left'},
                {type: 'main', style: 'background-color: white;', style: pstyle, overflow: 'hidden'},
                {type: 'preview', size: '50%', resizable: true, hidden: true, style: pstyle, content: 'preview'},
                {type: 'right', size: 200, resizable: true, style: pstyle, content: 'right'},
                {type: 'bottom', size: 50, resizable: true, hidden: true, style: pstyle, content: 'bottom'}
            ]
        });
    }

    $(function () {
        init_w2ui();
    });
 </script>
</html>
```

I get the following message in browser console

```javascript
Uncaught Error: Syntax error, unrecognized expression: #layout_layout_panel_function (symbol) {
      var $a, self = this, $iter = TMP_4.$$p, block = $iter || nil;

      var array_size = arguments.length - 1;
      if(array_size < 0) array_size = 0;
      var args = new Array(array_size);
      for(var arg_index = 0; arg_index < array_size; arg_index++) {
        args[arg_index] = arguments[arg_index + 1];
      }
      TMP_4.$$p = null;
      return $scope.get('Kernel').$raise($scope.get('NoMethodError').$new((function() {if ((($a = self.$inspect && !self.$inspect.$$stub) !== nil && (!$a.$$is_boolean || $a == true))) {
        return "undefined method `" + (symbol) + "' for " + (self.$inspect()) + ":" + (self.$$class)
        } else {
        return "undefined method `" + (symbol) + "' for " + (self.$$class)
      }; return nil; })(), symbol));
    } > .w2ui-panel-content
```

**Comment by bwl21** [comment]
@iliabylich wrote:
Well, the error comes from this place - https://github.com/vitmalina/w2ui/blob/1.4.0/src/w2layout.js#L745. The library iterates over an array using `for (var i in [])` and they don't check for if hasOwnProperty(i)

Opal adds further properties to Array, thus this "enumerating" pattern does not work and should be replaced by an iteration (http://web.archive.org/web/20101213150231/http://dhtmlkitchen.com/?category=/JavaScript/&date=2007/10/21/&entry=Iteration-Enumeration-Primitives-and-Objects)

@vitmalina could you fix that?

**Comment by bwl21** [comment]
I updated to the latest master  branch and it seems that this issue is already fixed.in commit 05bc53c

**Comment by bwl21** [comment]
@vitmalina I reopen this, just to request your advice what todo: as far as I can see, your fix is not merged into an official release. I can start to develop my project against your master branch, but I would like to rely on an official release. What can i do?

**Comment by mrg2001** [comment]
I think you are correct, for (var i in []) was fixed in current master, but not in 1.4.x branch. And it is unlikely it will be fixed there.

Master branch is next release, which is 1.5. I will be in charge of it. It is currently in feature complete stage and we are working on updating documentation, etc. No date for its release yet, but I think it is very stable and can be used instead of 1.4 branch.

**Comment by bwl21** [comment]
I switched one of my two projects to the masterbranch here. As 1.5 is not fully compatible to 1.4, I need to stick with this. Luckily this is not an Opal project
